### PR TITLE
Documentation update for 4.1.0 release

### PR DESCRIPTION
**Pull request by PravinPK**
## Description

Following updates are made to the documentation
- Reorganized the folder structure for markdown documents according to topics
- Documented PushTrackingStatus enum
- Documented "How to display Rich push notification"
- Separated In-App Messaging and Push Notification documents for easier accessibility.
- Remove the API description of deprecated API 
` Messaging.handleNotificationResponse(response, applicationOpened: true, customActionId: nil)`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Documentation/sources/push_notification/display-rich-push-notification.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# Display rich push notifications
+
+You must use Notification Service app extension to download images or other media attachments for the notification before displaying it on user's iOS device.
```
Comment (sbenedicadb) [inline Documentation/sources/push_notification/display-rich-push-notification.md:3]: `You must use a Notification Service app extension...`

File: Documentation/sources/push_notification/display-rich-push-notification.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+
+extension UNNotificationRequest {
+    var adobeAttachment: UNNotificationAttachment? {
+        // return nil if the notification does not contain a valid value for adb_media key
+        guard let attachmentString = content.userInfo["adb_media"] as? String else {
+            return nil
+        }
+        
+        // do not attach anything if its not a valid URL
```
Comment (sbenedicadb) [inline Documentation/sources/push_notification/display-rich-push-notification.md:53]: its > it is

File: Documentation/sources/troubleshooting/validate-messages-in-griffon.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Validate in-app messaging using AEPAssurance SDK extension and the Assurance UI
+# Validate In-App Messaging using Assurance
```
Comment (sbenedicadb) [inline Documentation/sources/troubleshooting/validate-messages-in-griffon.md:1]: in-app messaging shouldn't be capitalized, it's not a proper noun :)

**Review by sbenedicadb — CHANGES_REQUESTED** [review]
a couple small changes requested

File: Documentation/sources/troubleshooting/validate-messages-in-griffon.md (hunk 3 above)
Comment (PravinPK) [reply Documentation/sources/troubleshooting/validate-messages-in-griffon.md:1]: reverting 👍🏼

**Review by sbenedicadb — APPROVED** [review]

**Pull request merged**